src/HtmlSanitizer/HtmlSanitizerOptions.cs

```diff
@@ -0,0 +1,47 @@
+using AngleSharp.Css.Dom;
+using System;
+using System.Collections.Generic;
+
+namespace Ganss.XSS
+{
+    /// <summary>
+    /// Provides options to be used with <see cref="HtmlSanitizer"/>.
+    /// </summary>
+    public class HtmlSanitizerOptions
+    {
+        /// <summary>
+        /// Gets or sets the allowed tag names such as "a" and "div".
+        /// </summary>
+        public ISet<string> AllowedTags { get; set; } = new HashSet<string>();
```

Comment:
Shouldn't these have private setters? Also, they should be initialized with `StringComparer.OrdinalIgnoreCase`.

Reply:
No, I would want public setters because then I can easily assign a set to the properties using  object initializer syntax. Without a setter, I would have to loop over a set and append each item in the set using the `Add` method.

Yes, it should probably be initialized with `StringComparer.OrdinalIgnoreCase` if we want it to be case-insensitive, which we probably do want.

Reply:
You can use object and collection initializer syntax even if the setter is private. Collection initializer syntax works with the `Add()` method so you can do this:

```c#
public class SetTest
{
    public HashSet<string> Set { get; private set; } = new HashSet<string>();
}

var s = new SetTest()
{
    Set = { "a", "b" }
};
```

Reply:
Yes. but you cannot assign from a predefined set like this:
```cs
public static class Presets
{
    public static HashSet<string> Default { get; } = new HashSet<string>();
}

var s = new SetTest()
{
    Set = Presets.Default
};
```

Reply:
OK, then how about making them init-only?

Reply:
Good idea! I like that!

Reply:
There's [this problem](https://www.mking.net/blog/error-cs0518-isexternalinit-not-defined) now. Not sure how to deal with it or if it's even worth it.

Reply:
It seems easier to just skip `init` and just have public setters instead.

Reply:
Yeah ok, I agree.